JitsiConference.js

```diff
@@ -552,24 +500,98 @@ JitsiConference.prototype.removeTrack = function (track) {
         return Promise.reject(
             new JitsiTrackError(JitsiTrackErrors.TRACK_IS_DISPOSED));
     }
+    return this.replaceStream (track, null);
+};
 
-    if(!this.room){
-        if(this.rtc) {
-            this.onTrackRemoved(track);
+/**
+ * Replaces oldStream with newStream and performs a single offer/answer
+ *  cycle after both operations are done.  Either oldStream or newStream
+ *  can be null; replacing a valid 'oldStream' with a null 'newStream'
+ *  effectively just removes 'oldStream'
+ * @param oldStream the current stream in use to be replaced
+ * @param newStream the new stream to use
+ * @returns {Promise}
+ */
+JitsiConference.prototype.replaceStream = function (oldStream, newStream) {
```

Comment:
At list for all public APIs we have only the concept of tracks I think. And the objects that you are passing are of type JitsiLocalTrack so ...